### PR TITLE
Everything will now be checked before deleting (#152)

### LINKED ISSUE #152: Check if exists before deleting
# Description

# Suggestion
if (!JpaRepository.existsById(id)) return ControllerResult.of(Error.NOT_FOUND, NAME_KEY);

### PR DESCRIPTION
**Pull request by PAException**
## Description

Closes #152 

## Type of change

- [ ] Bug fix
- [x] Feature request
- [ ] Documenting code

## Checklists

- [x] I referenced the according Issue
  - [x] I created an Issue
  - [x] I put a link in the Section [Related Issue](#related-issue)
- [ ] I tested my code with multiple browsers
- [x] My commit messages are good and understandable

### CODE REVIEW

File: src/main/java/de/themorpheus/edu/taskservice/controller/ModuleController.java [hunk 1]
```diff
@@ -29,18 +31,22 @@
 		return ControllerResult.of(this.moduleRepository.getModuleByNameKeyIgnoreCase(nameKey));
 	}
 
-	public void deleteModule(String nameKey) {
+	public ControllerResult<Object> deleteModule(String nameKey) {
```
Comment (steve-hb) [inline src/main/java/de/themorpheus/edu/taskservice/controller/ModuleController.java:34]: Generic should contain specific type

File: src/main/java/de/themorpheus/edu/taskservice/controller/SubjectController.java [hunk 2]
```diff
@@ -28,11 +29,12 @@
 		return ControllerResult.of(this.subjectRepository.findAll());
 	}
 
-	public void deleteSubject(String nameKey) {
-		SubjectModel subjectModel = this.subjectRepository.getSubjectByNameKeyIgnoreCase(nameKey);
-		if (Validation.validateNull(subjectModel)) return;
+	public ControllerResult<Object> deleteSubject(String nameKey) {
```
Comment (steve-hb) [inline src/main/java/de/themorpheus/edu/taskservice/controller/SubjectController.java:32]: Use specific type in generic

File: src/main/java/de/themorpheus/edu/taskservice/controller/TaskTypeController.java [hunk 3]
```diff
@@ -3,25 +3,32 @@
...
 		return ControllerResult.of(this.taskTypeRepository.save(new TaskTypeModel(-1, nameKey)));
 	}
 
-	public ControllerResult<TaskTypeModel> getTaskTypeByNameKey(String nameKey) {
+	public ControllerResult<TaskTypeModel> getTaskType(String nameKey) {
 		return ControllerResult.of(this.taskTypeRepository.getTaskTypeByNameKeyIgnoreCase(nameKey));
 	}
 
-	public void deleteTaskType(String nameKey) {
+	public ControllerResult<Object> deleteTaskType(String nameKey) {
```
Comment (steve-hb) [inline src/main/java/de/themorpheus/edu/taskservice/controller/TaskTypeController.java:26]: Use specific generic type

File: src/main/java/de/themorpheus/edu/taskservice/controller/solution/MultipleChoiceSolutionController.java [hunk 4]
```diff
@@ -50,22 +50,18 @@
...
-	public void deleteAll(int taskId) {
-		ControllerResult<SolutionModel> optionalSolution = this.solutionController.getOrCreateSolution(taskId, NAME_KEY);
-		if (optionalSolution.isResultNotPresent()) return;
+		ControllerResult<SolutionModel> optionalSolution = this.solutionController.getSolution(taskId, NAME_KEY);
+		if (optionalSolution.isResultNotPresent()) return ControllerResult.ret(optionalSolution);
 
-		List<MultipleChoiceSolutionModel> multipleChoiceSolutionModels = this.multipleChoiceSolutionRepository.findAllById(
-				Collections.singleton(optionalSolution.getResult().getSolutionId()));
-		if (multipleChoiceSolutionModels.isEmpty()) return;
+		if (this.multipleChoiceSolutionRepository.existsById(optionalSolution.getResult().getSolutionId()))
```
Comment (steve-hb) [inline src/main/java/de/themorpheus/edu/taskservice/controller/solution/MultipleChoiceSolutionController.java:56]: Will this even work?

File: src/main/java/de/themorpheus/edu/taskservice/controller/solution/MultipleChoiceSolutionController.java [hunk 5]
```diff
@@ -83,4 +79,15 @@ public void deleteAll(int taskId) {
 		return ControllerResult.of(dto);
 	}
 
+	public void deleteAll(int taskId) {
+		ControllerResult<SolutionModel> optionalSolution = this.solutionController.getOrCreateSolution(taskId, NAME_KEY);
```
Comment (steve-hb) [inline src/main/java/de/themorpheus/edu/taskservice/controller/solution/MultipleChoiceSolutionController.java:83]: getOr**Create**?!

File: src/main/java/de/themorpheus/edu/taskservice/controller/solution/MultipleChoiceSolutionController.java [hunk 6]
```diff
@@ -83,4 +79,15 @@ public void deleteAll(int taskId) {
...
+		if (optionalSolution.isResultNotPresent()) return;
+
+		List<MultipleChoiceSolutionModel> multipleChoiceSolutionModels = this.multipleChoiceSolutionRepository.findAllById(
```
Comment (steve-hb) [inline src/main/java/de/themorpheus/edu/taskservice/controller/solution/MultipleChoiceSolutionController.java:86]: Same as in the other MR: Collection-usage is useless and we should use exist/count

**Review by steve-hb — CHANGES_REQUESTED** [review]

File: src/main/java/de/themorpheus/edu/taskservice/controller/solution/MultipleChoiceSolutionController.java (hunk 4 above)
Comment (PAException) [reply src/main/java/de/themorpheus/edu/taskservice/controller/solution/MultipleChoiceSolutionController.java:56]: yes

**Pull request merged**